privacyidea/lib/tokens/indexedsecrettoken.py

```diff
@@ -170,10 +170,14 @@ def update(self, param, reset_failcount=True):
         :return: nothing
 
         """
-        if 'genkey' not in param and 'otpkey' not in param:
+        if 'verify' not in param and 'genkey' not in param and 'otpkey' not in param:
             param['genkey'] = 1
 
         TokenClass.update(self, param, reset_failcount)
+
+        # finally we set the otplen of the token, no matter if the otpkey was generated or not
```

Comment:
why do we need to set the `otplen`? Technically the `otplen` for the indexed-secret token is defined in the policy `indexedsecret_count` or defaults to 2.

Reply:
You are right. I was mislead and confused by the logs from the ticket.

I confused the otplen (the count of characters input by the user) and the length of the secret.
I need to rethink this.

Reply:
Done in 0de5ec9e2